call-home/src/bin/callhome/collector/report_models.rs

```diff
@@ -25,8 +25,8 @@ pub(crate) struct Volumes {
 impl Volumes {
     /// Receives a openapi::models::Volumes object and returns a new report_models::volume object by
     /// using the data provided.
-    pub(crate) fn new(volumes: openapi::models::Volumes, event_data: EventData) -> Self {
-        let volumes_size_vector = get_volumes_size_vector(volumes.entries);
+    pub(crate) fn new(volumes: Vec<Volume>, event_data: EventData) -> Self {
```

Comment:
We should test this for memory use to see if we have enough on the memory resources limit but I think the limit seems enough

Reply:
It's 16 MiB to 32 MiB request and limit at the moment.

I'll test! This is my first time using valgrind in a 'real world' setting outside of cs107! I'll mock 10,000 Volumes across 1000 Nodes.

Reply:
I've tested this and there's about <41 MiB of heap usage.

I've tested with 10,000 volumes, 0 nodes, 0 pools (scale of the vols seemed to cover most typical nodes and pool counts as far as memory usage is concerned).

I'll bump the limit to 64 MiB.

Reply:
So iterator elements are copied. Fixing those allocations with iterator level manipulations should eliminate a lot of this memory usage.

Reply:
That should be plenty then, though if we reduce those copies then maybe we can keep it at 32

Reply:
The request is set to 16 MiB. So it always starts out at 16 MiB.